hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/FieldAccessExpression.java

```diff
@@ -25,23 +26,22 @@
 import com.hazelcast.sql.impl.QueryException;
 import com.hazelcast.sql.impl.row.Row;
 import com.hazelcast.sql.impl.type.QueryDataType;
+import org.apache.avro.generic.GenericRecord;
 
 import java.io.IOException;
-import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * An expression backing the DOT operator for extracting field from a struct type.
  * <p>
- * {@code ref.field} - extracts `field` from `ref`.
+ * {@code ref.field} extracts {@code field} from {@code ref}.
  */
 public class FieldAccessExpression<T> implements Expression<T> {
-    // FAE can be potentially used for many sub-classes of the base class, but it will always use same getter
+    // FAE can be potentially used for many subclasses of the base class, but it will always use same getter
     private static final int MAX_CLASS_COUNT = 10;
     private static final int MAX_GETTER_PER_CLASS_COUNT = 1;
 
     // single instance for all calls to eval, used only during execution on particular node
-    // atomic reference due to serialization constraints
-    private final AtomicReference<GetterCache> getterCache = new AtomicReference<>();
+    private transient GetterCache getterCache;
```

Comment:
@ivanthescientist "serialization constraints" what serialization constraints did you mean here?

Reply:
thanks. So the only reason to use `AtomicReference` was to leverage "conditional serializability" of it? I.e. if it contains null (before query execution), it is Serializable? If so, then `transient` will be much more explicit regarding the intent.

Reply:
is `FieldAccessExpression` instance reused my multiple threads if we have processor with localParallelism > 1? This might be not important for index scan (uses LP=1) but maybe is for others (IMap scan, Calc?)?

Reply:
`IMapSqlConnector.fullScanReader()` uses `RowProjectorProcessorSupplier` with default local parallelism, so `FieldAccessExpression` (actually all `Expression`s) must be thread-safe.